fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -578,58 +1007,114 @@ struct RangeFileWriter {
 	Key lastValue;
 };
 
+ACTOR static Future<Void> decodeKVPairs(StringRefReader* reader,
+                                        Standalone<VectorRef<KeyValueRef>>* results,
+                                        bool encryptedBlock,
+                                        Optional<Database> cx,
+                                        Reference<TenantEntryCache<Void>> tenantCache) {
+	// Read begin key, if this fails then block was invalid.
+	state uint32_t kLen = reader->consumeNetworkUInt32();
+	state const uint8_t* k = reader->consume(kLen);
+	results->push_back(results->arena(), KeyValueRef(KeyRef(k, kLen), ValueRef()));
+	state KeyRef prevKey = KeyRef(k, kLen);
+	state bool done = false;
+
+	// Read kv pairs and end key
+	while (1) {
+		// Read a key.
+		kLen = reader->consumeNetworkUInt32();
+		k = reader->consume(kLen);
+		// make sure that all keys in a block belong to exactly one tenant,
+		// unless its the last key in which case it can be a truncated (different) tenant prefix
+		if (encryptedBlock && g_network && g_network->isSimulated()) {
```

Comment:
do we want to do this block `only` for simulation and not otherwise?

Reply:
I mean we can if we want considering we don't have any SLA/SLO for backup 😂 . But it can be expensive since we need to go the TenantCache each time

Reply:
I think if you are referring to `non-tenant` key-lookup then we should be able to fix that usecase soon. But, in general I agree, as this is a sanity check, we can limit it to `simulation runs`. Thanks